pkg/domain/infra/abi/artifact.go

```diff
@@ -80,17 +81,45 @@ func (ir *ImageEngine) ArtifactPull(ctx context.Context, name string, opts entit
 	return nil, artStore.Pull(ctx, name, *pullOptions)
 }
 
-func (ir *ImageEngine) ArtifactRm(ctx context.Context, name string, _ entities.ArtifactRemoveOptions) (*entities.ArtifactRemoveReport, error) {
+func (ir *ImageEngine) ArtifactRm(ctx context.Context, name string, opts entities.ArtifactRemoveOptions) (*entities.ArtifactRemoveReport, error) {
+	var (
+		namesOrDigests []string
+	)
+	artifactDigests := make([]*digest.Digest, 0)
```

Comment:
This should be moved just above the loop over `namesOrDigests`  so you can preallocate the full length.
`make([]*digest.Digest, 0, len(namesOrDigests))`